snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/importer/Rf2TransactionContext.java

```diff
@@ -111,25 +111,32 @@ public void clearContents() {
 	
 	@Override
 	public long commit() {
-		throw new UnsupportedOperationException("TODO implement me");
+		throw new UnsupportedOperationException("Use the single supported commit(String) method");
 	}
 	
 	@Override
-	public long commit(String userId, String commitComment, String parentContextDescription) {
+	public long commit(String commitComment) {
 		final Set<String> idsToRegister = ImmutableSet.copyOf(newComponents.keySet().stream().filter(SnomedIdentifiers::isValid).iterator());
-		try {
-			// clear local cache before executing commit
-			newComponents = newHashMap();
-			LOG.info("Pushing changes: {}", commitComment);
-			long timestamp = getDelegate().commit(userId, commitComment, parentContextDescription);
-			// after successful commit register all commited IDs to CIS
-			final ISnomedIdentifierService cis = service(ISnomedIdentifierService.class);
-			if (cis.importSupported()) {
-				cis.register(idsToRegister);
-			}
-			return timestamp;
-		} finally {
+		// clear local cache before executing commit
+		newComponents = newHashMap();
+		LOG.info("Pushing changes: {}", commitComment);
+		long timestamp = getDelegate().commit(commitComment);
+		// after successful commit register all commited IDs to CIS
+		final ISnomedIdentifierService cis = service(ISnomedIdentifierService.class);
+		if (cis.importSupported()) {
+			cis.register(idsToRegister);
```

Comment:
Consider wrapping this part in a `try-catch` block (although not sure if we want to move forward with an import that has desynced from the CIS).

Reply:
Well, yeah, that's something that should not happen, otherwise duplicate content will start appearing and that's always a bad sign. The CIS registration logic has been simplified greatly to allow as much ID as possible in, so I think we are safe for now. 😄